### DISCUSSION TITLE
some wrong about "os error 2" of Lotus-miner

### DISCUSSION THREAD

**Discussion by shmilytsui** [filecoin-project/lotus#7994] [Storage Provider]
### Checklist

- [X] This is **not** a security-related bug/issue. If it is, please follow please follow the [security policy](https://github.com/filecoin-project/lotus/security/policy).
- [X] This is **not** a question or a support request. If you have any lotus related questions, please ask in the [lotus forum](https://github.com/filecoin-project/lotus/discussions).
- [X] This is **not** a new feature request. If it is, please file a [feature request](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Ffeature&template=feature_request.yml) instead.
- [X] This is **not** an enhancement request. If it is, please file a [improvement suggestion](https://github.com/filecoin-project/lotus/issues/new?assignees=&labels=need%2Ftriage%2Ckind%2Fenhancement&template=enhancement.yml) instead.
- [X] I **have** searched on the [issue tracker](https://github.com/filecoin-project/lotus/issues) and the [lotus forum](https://github.com/filecoin-project/lotus/discussions), and there is no existing related issue or discussion.
- [X] I am running the [`Latest release`](https://github.com/filecoin-project/lotus/releases), or the most recent RC(release canadiate) for the upcoming release or the dev branch(master), or have an issue updating to any of these.
- [X] I did not make any code changes to lotus.

### Lotus component

- [ ] lotus daemon - chain sync
- [ ] lotus miner - mining and block production
- [ ] lotus miner/worker - sealing
- [ ] lotus miner - proving(WindowPoSt)
- [ ] lotus miner/market - storage deal
- [ ] lotus miner/market - retrieval deal
- [ ] lotus miner/market - data transfer
- [ ] lotus client
- [ ] lotus JSON-RPC API
- [ ] lotus message management (mpool)
- [ ] Other

### Lotus Version

```text
[Version]
lotus        version 1.13.2+mainnet+git.f9db71ee1.dirty
lotus-miner  version 1.13.2+mainnet+git.f9db71ee1.dirty
lotus-worker version 1.13.2+mainnet+git.f9db71ee1.dirty
```

### Describe the Bug

when I was trying to test "lotus-miner sectors pledger" , the miner-log print out some errors, and the worker-log print out some warn:
This is my software development environment:
[Setup]
CPU: 32  AMD Ryzen 9 3950X 16-Core Processor
RAM:  Range Size: 128 GB
Boot: Samsung SSD 860
GPU:  NVIDIA Corporation TU102 [GeForce RTX 2080 Ti] (rev a1)

[lotus-miner info]
Enabled subsystems (from miner API): [Mining Sealing SectorStorage Markets]
Enabled subsystems (from markets API): [Mining Sealing SectorStorage Markets]
Chain: [sync ok] [basefee 106 aFIL]
Miner: f01698366 (32 GiB sectors)
Power: 0  / 15.1 Ei (0.0000%)
        Raw: 0 B / 15.02 EiB (0.0000%)
        Committed: 0 B
        Proving: 0 B
Below minimum power threshold, no blocks will be won
Miner Balance:    0
      PreCommit:  0
      Pledge:     0
      Vesting:    0
      Available:  0
Market Balance:   0
       Locked:    0
       Available: 0
Worker Balance:   149.997 mFIL
Total Spendable:  149.997 mFIL

Sectors:
        Total: 10
        Packing: 1
        Removed: 9

Storage Deals: 0, 0 B

Retrieval Deals (complete): 0, 0 B

[lotus-miner actor control list]
$ lotus-miner actor control list
name    ID         key           use    balance                   
owner   f01698347  f3rqznipn...  other  0.149996565248101503 FIL  
worker  f01698347  f3rqznipn...  other  0.149996565248101503 FIL 

[storage.json]
$ cat storage.json 
{
  "StoragePaths": [
    {
      "Path": "/data/seal"
    },
    {
      "Path": "/data/store"
    }
  ]
}

[lotus-miner storage list]
$ lotus-miner storage list
060c7711-83b4-40ea-8e27-8f4a2582a0b0:
        [##########################################        ] 379.8 GiB/441.9 GiB 85%
        Unsealed: 0; Sealed: 0; Caches: 0; Reserved: 0 B
        Weight: 10; Use: Store
        Local: /data/store
        URL: http://192.168.1.10:2345/remote

40a92049-54da-4e73-a2a9-b9990f64b1b2:
        [##########################################        ] 379.8 GiB/441.9 GiB 85%
        Unsealed: 0; Sealed: 0; Caches: 0; Reserved: 0 B
        Weight: 10; Use: Seal 
        Local: /data/seal
        URL: http://192.168.1.10:2345/remote

68202d09-70e4-431d-a1c2-483190e84e27:
        [##########################################        ] 379.8 GiB/441.9 GiB 85%
        Unsealed: 1; Sealed: 0; Caches: 0; Reserved: 0 B
        Weight: 10; Use: Seal 
        URL: http://192.168.1.10:3456/remote (latency: 700µs)

b684eebe-d68c-4fdd-8c6b-a01b559e814f:
        [##########################################        ] 379.8 GiB/441.9 GiB 85%
        Unsealed: 0; Sealed: 0; Caches: 0; Reserved: 0 B
        Weight: 10; Use: Store
        URL: http://192.168.1.10:3456/remote (latency: 600µs)

### Logging Information

```text
[lotus-miner log]:
2022-01-23T08:47:42.255+0800    ERROR   sectors storage-sealing/fsm.go:28       unhandled sector error (8): filling up the sector ([34091302912]):
    github.com/filecoin-project/lotus/extern/storage-sealing.(*Sealing).handlePacking
        /opt/src/git-lotus/lotus/extern/storage-sealing/states_sealing.go:77
  - add piece:
    github.com/filecoin-project/lotus/extern/storage-sealing.(*Sealing).padSector
        /opt/src/git-lotus/lotus/extern/storage-sealing/states_sealing.go:96
  - storage call error 0: generating piece commitment: failed to build tree

Caused by:
    0: failed to create data store
    1: No such file or directory (os error 2)	

[lotus-worker log]:
2022-01-24T01:08:20.100+0800    WARN    ffi-wrapper     ffiwrapper/ffiwrapper.go:52     copied different amount than expected: 65536 != 4161536
```

### Repo Steps

1. Run '...'
2. Do '...'
3. See error '...'
...

**Comment by Reiers** [comment]
Hi @shmilytsui 

Thanks for the report. The issue you are experiencing lies elsewhere.

Moving this issue to Lotus Discussions for help and troubleshooting.

Closing Lotus issue ticket.

**Comment by SgtCoin** [comment]
Hi @shmilytsui 

We can spot check a few things here to ensure the basics are taken care of. 

1. What is your TMPDIR environment variable set to? 
2. Did anything cause an interruption to your storage? 

Did this continue to happen with newly pledge sectors or did only occur once?

**Reply by shmilytsui** [reply]
Thank you!
I didn't found any wrong before I run "lotus-miner sectors pledges"
this is my script about lotus,lotus-miner and lotus-worker:
[lotus]
$ cat /usr/local/bin/lotus
IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/ \
LOTUS_PATH=/shared/lotus \
/opt/local/lotus/lotus $*

[lotus-miner]
$ cat /usr/local/bin/lotus-miner
TMPDIR=/data/tmp \
FIL_PROOFS_PARAMETER_CACHE=/data/filecoin-proof-parameters \
IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/ \
LOTUS_PATH=/data/lotus \
LOTUS_MINER_PATH=/data/lotus-miner  \
FIL_PROOFS_USE_GPU_COLUMN_BUILDER=1 \
FIL_PROOFS_USE_GPU_TREE_BUILDER=1 \
FIL_PROOFS_MAXIMIZE_CACHING=1 \
RUST_LOG=Info \
/opt/local/lotus/lotus-miner $*

[lotus-miner]
$ cat /usr/local/bin/lotus-worker
TMPDIR=/data/tmp \
FIL_PROOFS_PARAMETER_CACHE=/data/filecoin-proof-parameters \
FIL_PROOFS_PARENT_CACHE=/data/filecoin-parents \
IPFS_GATEWAY=https://proof-parameters.s3.cn-south-1.jdcloud-oss.com/ipfs/ \
LOTUS_WORKER_PATH=/data/lotus-worker \
LOTUS_MINER_PATH=/data/lotus-miner  \
FIL_PROOFS_USE_GPU_COLUMN_BUILDER=1 \
FIL_PROOFS_USE_GPU_TREE_BUILDER=1 \
FIL_PROOFS_MAXIMIZE_CACHING=1 \
FIL_PROOFS_USE_MULTICORE_SDR=1 \
RUST_LOG=Info \
BELLMAN_CPU_UTILIZATION=0.875 \
/opt/local/lotus/lotus-worker $*
